app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,48 @@
+class CommentsController < ApplicationController
+  before_action :authenticate_user!, only: [:edit, :update, :destroy, :create, :new]
+
+  def show
+    @post = Post.find(params[:post_id])
+    @comment = Comment.find(params[:id])
+  end
+
+  def edit
+    @post = Post.find(params[:post_id])
+    @comment = Comment.find(params[:id])
+  end
+
+  def new
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.new
+  end
+
+  def update
+    @post = Post.find(params[:post_id])
+    @comment = Comment.find(params[:id])
+    @comment.update(comments_params)
+
+    redirect_to post_comment_path(@post, @comment)
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.create(comments_params)
+    redirect_to @post
+  end
+
+  def destroy
+    @post = Post.find(params[:post_id])
```

Comment:
Try adding in a private `set_post` method to run before each action so you don't have to write this line a zillion times.